### PR TITLE
fix: upgrade version to manage authentication in US

### PR DESCRIPTION
**Pull request by johannes-lindgren**
Issue: EXT-1541

## What

Upgrading the Nextjs starter to be compatible with US spaces:

1. Upgrading the version of @storyblok/app-extension-auth
2. Using `storyblok-js-client` to fetch data, because it has an option to provide the region.
3. Increased type safety by validating responses.

## How to test

Try this app on a EU space and a US space:

Install with

https://app.storyblok.com/#!/install/storyblok-gmbh@app-extension-nextjs-starter

Open the Vercel preview deployment for this PR by appending `?dev=1` to the URL.

You should see something like

image

### CODE REVIEW

**Comment by eunjae-lee** [comment]
Not sure if this PR is ready for review (because it says it's a draft PR). Ping me when you think it's ready (and also when it passes the CI)

File: app-nextjs-starter/pages/api/stories.ts [hunk 1]
```diff
@@ -1,8 +1,32 @@
...
+    region,
+  }).get(
+    `spaces/${spaceId.toString(10)}/stories`
+  ).then(res =>
+    res.data as unknown
+  ).then(data =>
+    isStoriesResponse(data) ? data.stories : undefined
+  ).catch((error) =>
+    error
+  )
```
Comment (eunjae-lee) [inline app-nextjs-starter/pages/api/stories.ts:27]: With this, `fetchStories` will always resolve but I saw a code like

```js
  useEffect(() => {
    fetchStories(props.spaceId, props.userId)
      .then(stories => setStories(stories))
  }, [props.spaceId, props.userId])
```

then that `stories` might be either real `stories` or `error`. I guess you meant to print out error and return undefined or something like that?

**Review by eunjae-lee — CHANGES_REQUESTED** [review]

File: app-nextjs-starter/pages/api/stories.ts (hunk 1 above)
Comment (johannes-lindgren) [reply app-nextjs-starter/pages/api/stories.ts:27]: It's a different `fetchStories`... one is for the client and the other for the server.

Comment (eunjae-lee) [reply app-nextjs-starter/pages/api/stories.ts:27]: Ahah you're right. I see which pattern you used here (following up our conversation yesterday 🙂)

**Review by eunjae-lee — APPROVED** [review]

**Pull request merged**